counsel.el

```diff
@@ -5087,6 +5088,30 @@ this if you want to provide specific non-standard build types to
 `counsel-compile'.  The default helpers are setup to handle common
 build environments.")
 
+;; Should this be hook-able to deal with project managers and other
+;; version control systems?
+(defun counsel--find-project-root ()
+  "Traverse up the tree and find the project root."
+  (expand-file-name (locate-dominating-file
+                     (or (buffer-file-name) default-directory)
+                     ".git")))
```

Comment:
At least two other places exist, which are trying to do something more or less similar,

https://github.com/abo-abo/swiper/blob/2a02343cdf218441158e83323ce507a6d9112445/counsel.el#L1166-L1169

https://github.com/abo-abo/swiper/blob/2a02343cdf218441158e83323ce507a6d9112445/counsel.el#L2628-L2630

and it would be nice to bring all of them to a common ground via some central function, which is also hookable as per your comment above since `".git"` looks too opinionated albeit satisfying in the majority of cases.

Reply:
Good idea. I guess if I expand counsel-locate-git-root with an &optional dir and adjust the others appropriately?